static/js/controllers/reports.js

```diff
@@ -250,17 +335,18 @@ var _ = require('underscore'),
             _query();
           }
         }
-
       };
 
       $scope.$on('ClearSelected', function() {
-        $scope.selected = null;
+        $scope.selected = [];
+        $('#reports-list input[type="checkbox"]')
+          .prop('checked', false);
         liveList.clearSelected();
```

Comment:
Yeah, there are a bunch of functions, it does add a bunch of code, it doesn't reduce it. But it's cleaner, better organized code, easier to understand and modify. It does add a layer of abstraction, and the point of abstraction is to do exactly that!

There's basically select/unselect by id, select/unselect all, that's 3 functions, it's not long to write out fully. (am I being optimistic? :D)

While you're at it you could also package in the modification of `$scope.selected` (unless there's a reason it shouldn't match the livelist selection and the checkboxes).